app/browser/api/ledger.js

```diff
@@ -2419,6 +2407,34 @@ const initialize = (state, paymentsEnabled) => {
   }
 }
 
+const schedulePromoRefFetch = (state) => {
+  if (promoRefFetchTimeoutId) {
+    clearTimeout(promoRefFetchTimeoutId)
+  }
+
+  promoRefFetchTimeoutId = setTimeout(() => {
+    if (updateState.getUpdateProp(state, 'referralDownloadId') == null) {
```

Comment:
state in this case could be already outdated. You have up to 70s old state here

Reply:
App action has been created to remedy this, a fresh state will be used in these actions now.